modules/apps/headless/headless-object/headless-object-impl/src/main/java/com/liferay/headless/object/internal/dto/v1_0/converter/ObjectEntryFolderDTOConverter.java

```diff
@@ -43,6 +43,18 @@ public ObjectEntryFolder toDTO(DTOConverterContext dtoConverterContext)
 			_objectEntryFolderLocalService.getObjectEntryFolder(
 				(Long)dtoConverterContext.getId());
 
+		com.liferay.object.model.ObjectEntryFolder parentObjectEntryFolder =
+			null;
+
+		if (objectEntryFolder.getParentObjectEntryFolderId() > 0L) {
+			parentObjectEntryFolder =
+				_objectEntryFolderLocalService.getObjectEntryFolder(
+					objectEntryFolder.getParentObjectEntryFolderId());
+		}
+
+		com.liferay.object.model.ObjectEntryFolder
+			finalParentObjectEntryFolder = parentObjectEntryFolder;
```

Comment:
Instead of creating these two variables, can you extract this to private method?
```
ObjectEntryFolder parentObjectEntryFolder = _getParentObjectEntryFolder(objectEntryFolder);
...
private ObjectEntryFolder _getParentObjectEntryFolder(...) {
    if (objectEntryFolder.getParentEntryFolder() == ObjectEntryFolderConstants._..._) {
        return _objectEntryFolderLocalService.getObjectEntryFolder(...);
    }
    ...
}
```
This way, the variable will be effectively final and no extra one will be necessary.